app/Http/Controllers/Api/V1/TimeEntryController.php

```diff
@@ -115,6 +114,190 @@ public function index(Organization $organization, TimeEntryIndexRequest $request
+        $filter->addActiveFilter($request->input('active'));
+        $filter->addMemberIdFilter($member);
+        $filter->addMemberIdsFilter($request->input('member_ids'));
+        $filter->addProjectIdsFilter($request->input('project_ids'));
+        $filter->addTagIdsFilter($request->input('tag_ids'));
+        $filter->addTaskIdsFilter($request->input('task_ids'));
+        $filter->addClientIdsFilter($request->input('client_ids'));
+        $filter->addBillableFilter($request->input('billable'));
+        $timeEntriesQuery = $filter->get();
+
+        $user = $this->user();
+
+        /** @var string|null $group1Type */
+        $group1Type = $request->get('group');
+        /** @var string|null $group2Type */
+        $group2Type = $request->get('sub_group');
+
+        $group1Select = null;
+        $group2Select = null;
+        $groupBy = null;
+        if ($group1Type !== null) {
+            $group1Select = $this->getGroupByQuery($group1Type, $user->timezone, $user->week_start);
+            $groupBy = ['group_1'];
+            if ($group2Type !== null) {
+                $group2Select = $this->getGroupByQuery($group2Type, $user->timezone, $user->week_start);
+                $groupBy = ['group_1', 'group_2'];
+            }
+        }
+
+        $timeEntriesQuery->selectRaw(
+            ($group1Select !== null ? $group1Select.' as group_1,' : '').
+            ($group2Select !== null ? $group2Select.' as group_2,' : '').
+            ' round(sum(extract(epoch from (coalesce("end", now()) - start)))) as aggregate,'.
+            ' round(
+                  sum(
+                      extract(epoch from (coalesce("end", now()) - start)) * (coalesce(billable_rate, 0)::float/60/60)
+                  )
+              ) as cost'
+        );
+        if ($groupBy !== null) {
+            $timeEntriesQuery->groupBy($groupBy);
+        }
+
+        $timeEntriesAggregates = $timeEntriesQuery->get();
+
+        if ($group1Select !== null) {
+            $groupedAggregates = $timeEntriesAggregates->groupBy($group2Select !== null ? ['group_1', 'group_2'] : ['group_1']);
+
+            $group1Response = [];
+            $group1ResponseSum = 0;
+            $group1ResponseCost = 0;
+            foreach ($groupedAggregates as $group1 => $group1Aggregates) {
+                /** @var string $group1 */
+                $group2Response = [];
+                if ($group2Select !== null) {
+                    $group2ResponseSum = 0;
+                    $group2ResponseCost = 0;
+                    foreach ($group1Aggregates as $group2 => $aggregate) {
+                        /** @var string $group2 */
+                        /** @var Collection<int, object{aggregate: int, cost: int}> $aggregate */
+                        /** @var string $group2Type */
+                        $group2Response[] = [
+                            'type' => $group2Type,
+                            'value' => $group2 === '' ? null : $group2,
```

Comment:
i don't really have a better suggestion tbh but 'value' feels weird here. under 'value' i would assume the seconds. maybe we could do 'key'? or 'label'?

Reply:
I changed it to `key`